src/common/acl_principal.c

```diff
@@ -309,3 +311,116 @@ daos_acl_principal_to_gid(const char *principal, gid_t *gid)
 	D_FREE(buf);
 	return rc;
 }
+
+static bool
+is_special_type(enum daos_acl_principal_type type)
+{
+	switch (type) {
+	case DAOS_ACL_OWNER:
+	case DAOS_ACL_OWNER_GROUP:
+	case DAOS_ACL_EVERYONE:
+		return true;
+	case DAOS_ACL_USER:
+	case DAOS_ACL_GROUP:
+	default:
+		break;
+	}
+
+	return false;
```

Comment:
It might be clearer to make this more explicit by placing it inside the default as the return instead of breaking and going to the end of the function.